src/Rules/Actions.hs

```diff
@@ -126,6 +128,17 @@ copyDirectory source target = do
     putProgressInfo $ renderAction "Copy directory" source target
     quietly $ cmd cmdEcho ["cp", "-r", source, target]
 
+-- | Like copyDirectory, but with a predicate to decide which files/directories
+-- to include.
+copyDirectory' :: (FilePath -> IO Bool) -> FilePath -> FilePath -> Action ()
+copyDirectory' test source target = do
```

Comment:
I will rework this function because it has a flaw, it will recreate the full path that was given as source in the destination directory.